service/controller/resource/values/delete.go

```diff
@@ -0,0 +1,72 @@
+package values
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/annotation"
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	controllerkey "github.com/giantswarm/config-controller/service/controller/key"
+)
+
+func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
+	app, err := controllerkey.ToAppCR(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	configVersion, ok := app.GetAnnotations()[annotation.ConfigVersion]
+	if !ok {
+		r.logger.Debugf(ctx, "App CR %q is missing %q annotation", app.Name, annotation.ConfigVersion)
+		r.logger.Debugf(ctx, "cancelling resource")
+		return nil
+	}
+
+	r.logger.Debugf(ctx, "deleting App %#q, config version %#q", app.Spec.Name, configVersion)
+
+	if app.Spec.Config.ConfigMap.Name != "" {
```

Comment:
I think it should attempt the CM/Secret deletion regardless of what we have in App CR.

Reply:
Agreed. I updated the code to do that (we need to generate expected names if we've found none).